src/python/WMCore/JobStateMachine/ChangeState.py

```diff
@@ -302,7 +305,23 @@ def recordInCouch(self, jobs, newstate, oldstate, updatesummary = False):
                 logging.debug("Updated job summary state history for job %s" % jobSummaryId)
 
             if job.get("fwjr", None):
-
+                cached_task = self.task_cache.get(job['task'])
+                if cached_task is None:
+                    spec_path = self.getWorkflowSpecDAO.execute(job['task'])['spec']
```

Comment:
if I were to map the workflow then I would have to have an inner mapping of task to prepid i think